### ISSUE TITLE
IR codes database

### ISSUE THREAD

**Issue by tatunts** [dimagoltsman/Random-Broadlink-RM-Code-Generator#2]
Hello again @dimagoltsman!

I have another idea/suggestion for you: Broadlink's devices are very popular now. I believe that the Mini is the best solution for IR on the market right now, specially to work with Home Assistant. I don't know if you use it for IR often but I never found a way to set my devices IR codes without having to learn them one by one. Also, there are a lot of functions (IR codes) that are not available to learn from previously own controllers, for example, the command "change input to live TV" on my 2014 webos LG TV with magic remote. The other day I read somewhere that there is a ON and OFF separate IR commands for my TV (not power toggle command), that would be great to send a specific command to my TV (since there is no wake on lan feature).

I've found this amazing IR database site: http://www.remotecentral.com/cgi-bin/codes/

Is there a way to import/convert those codes to use them on our HA configs and send with BL Mini?

I believe that would help many users to discover new commands and speed up the configuration of new devices.

Thanks!

**Comment by dimagoltsman** [comment]
nice one, i will try to figure it out if i can convert it, but i dont have any devices from that lis. 
so i need help from you. i need you to send me some device you have from that list, and its code, i will try to understand and convert it and ill send you back the broadlink code for it so you can tell me if it works. how is that sounds to you?

**Comment by tatunts** [comment]
Yes! Great idea. I will find a matching IR code (my config and public database) and send both to you ok? Thanks!

**Comment by dimagoltsman** [comment]
yup... waiting

**Comment by tatunts** [comment]
Ok, so after a few minutes searching the web, I found this site that claims it is the largest IR code database available: http://irdb.tk/

I am sending you my Sony Receiver codes:
```
home_theater_sala_mute: JgBmAE8UEhUVEioSEhQnFBYSFRIVEhUSFRITFSYVKRIWEhMAAtZQExUSFRIoExQTKRIWEhQUEhQTFBMUGBIoEyYVExQTAALYUBMTExMUKhEUFCYUFhIVEhUSFRMTExYSJxQmFRUSFgANBQAA==
home_theater_sala_volup: JgCIAFASFRMmFRMUExQpEhUSFRMVExMUExQTFCkSKRIXERUAAtVOFBYSKBMTFBMUKRIVEhUTFhITExUSExQpEigTFRMVAALVTxQTFCoRFRIVEigUFBIVExYRFRIVEhUSKRIpEhQUFwAC004VFBMqERUSFRIpEhUSFhIVExMTExQTFCkSKRIUFBYADQU==
home_theater_sala_voldown: JgCIAFATKhApEhUSExUqERUTExMVEhcQFRMTFCoRKRIVEhcAAsBOFCoSJhQTFBYSKRMSFBcQFRITFBcRFRInFCoRFRMUAALCUBInFCoSEhYSFCgTFxAVEhUSFxEVEhUSKhEpEhUSEwACxFASKxAnFBUTFRIoExUSFRITFBMVFhEWEiYUKRIVEhcADQU==
home_theater_sala_power: JgBmAFATKBMVEikSFRIpEhUSFhEWEhUSFRIVEikSKRIWERYAAsFQEikSFhIpEhUSKRIVEhUSFhEWEhUSFRIpEikSFRIWAALBUBIpEhYRKRIWEikSFRIVEhUSFhEWEhUSKRIpEhUSFQANBQAA==
```
Here are the DB codes in HEX:
```
VOLUME MUTE: 0000 0068 0000 000D 0060 0018 0018 0018 0018 0018 0030 0018 0018 0018 0030 0018 0018 0018 0018 0018 0018 0018 0018 0018 0018 0018 0018 0018 0030 0420
VOLUME +:0000 0068 0000 000D 0060 0018 0018 0018 0030 0018 0018 0018 0018 0018 0030 0018 0018 0018 0018 0018 0018 0018 0018 0018 0018 0018 0018 0018 0030 0420
VOLUME -:0000 0068 0000 000D 0060 0018 0030 0018 0030 0018 0018 0018 0018 0018 0030 0018 0018 0018 0018 0018 0018 0018 0018 0018 0018 0018 0018 0018 0030 0408
POWER: 0000 0068 0000 000D 0060 0018 0030 0018 0018 0018 0030 0018 0018 0018 0030 0018 0018 0018 0018 0018 0018 0018 0018 0018 0018 0018 0018 0018 0030 0408
```

**Comment by tatunts** [comment]
It would be great to have those for my Sony "dumb" receiver hehe:

```
POWER OFF: 0000 0068 0000 0010 0060 0018 0030 0018 0030 0018 0030 0018 0030 0018 0018 0018 0030 0018 0018 0018 0018 0018 0018 0018 0018 0018 0018 0018 0030 0018 0030 0018 0018 0018 0018 0330
POWER ON: 0000 0068 0000 0010 0060 0018 0018 0018 0030 0018 0030 0018 0030 0018 0018 0018 0030 0018 0018 0018 0018 0018 0018 0018 0018 0018 0018 0018 0030 0018 0030 0018 0018 0018 0018 0348
```

**Comment by dimagoltsman** [comment]
can you try this code?
JgBmAAAAAGgAAAANAGAAGAAYABgAGAAYADAAGAAYABgAMAAYABgAGAAYABgAGAAYABgAGAAYABgAGAAYADAEIA0FAAA=
its home_theater_sala_mute

**Comment by tatunts** [comment]
I can do it in an hour, is that OK? I am not at home now...

**Comment by dimagoltsman** [comment]
no hurry :)

**Comment by tatunts** [comment]
Hello @dimagoltsman! I've just tested it. Nothing happens, not even the RM Mini lights up with that code :(

**Comment by dimagoltsman** [comment]
ok ill try figure it out when ill get home

**Comment by tatunts** [comment]
Ok. I've tested with my mute code and the mini lights up and de IR code is sent. Thanks!

**Comment by dimagoltsman** [comment]
just to be sure, try again  this on: JgBmAAAAAGgAAAANAGAAGAAYABgAGAAYADAAGAAYABgAMAAYABgAGAAYABgAGAAYABgAGAAYABgAGAAYADAEIA0FAAAAAAAA

**Comment by tatunts** [comment]
No sorry... Nothing happens on the Mini.

**Comment by dimagoltsman** [comment]
JgA8AAAAAGgAAAANAGAAGAAYABgAGAAYADAAGAAYABgAMAAYABgAGAAYABgAGAAYABgAGAAYABgAGAAYADAEIA==


?

**Comment by tatunts** [comment]
Still nothing... Sorry.

**Comment by dimagoltsman** [comment]
Jg88AAAAAGgAAAANAGAAGAAYABgAGAAYADAAGAAYABgAMAAYABgAGAAYABgAGAAYABgAGAAYABgAGAAYADAEIA==

?

**Comment by tatunts** [comment]
Sorry I had to left home haha 

Will try and get back to you ASAP, OK? Thanks!

**Comment by dimagoltsman** [comment]
sure....

**Comment by tatunts** [comment]
Well, I do have a RM PRO here at work and I've tested your code and nothing happens... Sorry.

**Comment by dimagoltsman** [comment]
what do u mean nothing? it does send a signal from my broadlink

**Comment by tatunts** [comment]
No, the light on the RM PRO does not flashes so I don't believe that the IR code is sent. If I use my mute code on the same RM, the light at least flashes, thus sending the code.

**Comment by dimagoltsman** [comment]
it does flashes on mine :(

**Comment by tatunts** [comment]
this is odd, I will check it, mine doesn't do anything... Again, if use the same send_package data but paste my code, the code is sent. With yours, nothing happens... :(

**Comment by tatunts** [comment]
Ok, so I had to look really close to the LED on the RM. It actually flashes so quickly that I can hardly see it! It is like your RF codes with repetitions: of no repeat is set the light flashes very quickly, if is set to 2 or more, it takes longer! If I use my IR code, the light flashes for longer. Very odd, indeed! I will try to film it, so you can see! haha

**Comment by dimagoltsman** [comment]
i know about the repeat....
try this:

Jg88ABdoAA1gGBgYGBgwGBgYMBgYGBgYGBgYGBgYGBgwBCANBQ==

**Comment by tatunts** [comment]
Please take a look: https://photos.app.goo.gl/nrQYPGUqyrQSIeDA2

**Comment by tatunts** [comment]
With your last code, the IR code is sent for about 4 seconds....

**Comment by dimagoltsman** [comment]
yes i increased the repeat, but did it mute the device?

**Comment by tatunts** [comment]
Sorry, I am not at home now... I was trying with my RM PRO on my office. I also have my HASS configured here :) I will get back to you when I arrive home, ok. Thanks!

**Comment by dimagoltsman** [comment]
cool

**Comment by dimagoltsman** [comment]
ok, forget everything!
i understood the protocol from your database.
im 100% sure this will work:

JgAcAE8TExMTEycTExMnExMTExMTExMTExMTEycAA2YNBQAAAAAAAAAAAAA=

**Comment by tatunts** [comment]
Awesome! Will try as soon as I arrive home. I did preliminary test here and light flashes a bit quickly than I am used to, but I am confidant it will work too! Thanks my friend!

**Comment by dimagoltsman** [comment]
i found a converter from this code to broadlink, you can use this to connvert (remove spaces first):

https://gist.github.com/appden/42d5272bf128125b019c45bc2ed3311f

and convert the hex code you get from there to base64 with this:
http://tomeko.net/online_tools/hex_to_base64.php?lang=en

**Comment by tatunts** [comment]
Thanks for the information. I will try as soon as possible and get back to you. Thanks!
